projects/tsq-gallery/src/lib/components/tsq-gallery-viewer/tsq-gallery-viewer.component.ts

```diff
@@ -60,192 +53,152 @@ export class TSqGalleryViewerComponent {
   touchTimeStamp = 0;
   touchX = 0;
   initialTouchX = 0;
+  imageRotation = 0;
+  imageZoom = 1;
 
+  canGoForward = false;
+  canGoBack = false;
+  canZoomIn = false;
+  canZoomOut = false;
+  canMove = false;
+  isViewerOpen = false;
+  expanded = false;
   navigationDelay = false;
   pdfLoading = false;
-  keypress = false;
   isMoving = false;
-  imageRotation = 0;
-  imageZoom = 1;
+  galleryFiles: TSqGalleryFileModel[];
   selectedFileIndex: number;
+  selectedFileToDisplay: TSqGalleryFileModel;
+  topViewerContext: TSqGalleryTopViewerTemplateRefContext;
+  bottomViewerContext: TSqGalleryBottomViewerTemplateRefContext;
   scrollListener: () => void;
 
-  @ViewChild('backdrop', {static: false}) backdropRef: ElementRef;
+  constructor(private renderer: Renderer2) {}
 
-  private readFiles: TSqGalleryFileModel[];
-  private isOpen: boolean;
+  open(index?: number) {
+    this.isViewerOpen = true;
+    this.setCurrentFile(index);
 
-  constructor(private renderer: Renderer2) { }
+    setTimeout(() => {
+      if (!!this.backdropRef) {
+        this.backdropRef.nativeElement.focus();
 
-  contextClose = () => {
-    this.close();
+        const throttleFunc = throttle(this.onWindowScroll, 1);
+        this.scrollListener = this.renderer.listen(this.backdropRef.nativeElement, 'wheel', throttleFunc);
+      }
+    }, 10);
```

Comment:
this setTimeout exists to guarantee nativeElement existance?

Reply:
Yes